src/hooks/UseTasksStore.js

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from "react";
+import { useRecoilState } from "recoil";
+import { ListOfTasks } from "../store/store";
+
+export const useTasksStore = (taskId, item) => {
+  const [tasksList, setTasksList] = useRecoilState(ListOfTasks);
+  const index = useMemo(
+    () => tasksList.findIndex((task) => task.id === taskId),
+    [tasksList]
+  );
```

Comment:
Here you find `item` index, thus, you don't actually need to pass it to the hook. It should be:

```jsx
export const useTasksStore = (taskId) => {
```